numpy/lib/arraysetops.py

```diff
@@ -544,6 +545,26 @@ def in1d(ar1, ar2, assume_unique=False, invert=False):
         False where an element of `ar1` is in `ar2` and True otherwise).
         Default is False. ``np.in1d(a, b, invert=True)`` is equivalent
         to (but is faster than) ``np.invert(in1d(a, b))``.
+    kind : {None, 'sort', 'table'}, optional
+        The algorithm to use. This will not affect the final result,
+        but will affect the speed and memory use. The default, None,
+        will select automatically based on memory considerations.
+
+        * If 'sort', will use a mergesort-based approach. This will have
+          a memory usage of roughly 6 times the sum of the sizes of
+          `ar1` and `ar2`, not accounting for size of dtypes.
+        * If 'table', will use a key-dictionary approach similar
```

Comment:
If you can think of another word for `key-dictionary` that might be nice.  "lookup table"?

Reply:
It could be cool to shorten this and move some of the informatoin to the notes (even a table like the `np.sort` documentation).  But I think this is good for now.